src/security/ldap/src/main/java/org/geoserver/security/ldap/LDAPRoleService.java

```diff
@@ -222,7 +222,7 @@ public void executeWithContext(DirContext ctx,
                         String dn = LDAPUtils
                                 .getLdapTemplateInContext(ctx, template)
                                 .searchForSingleEntry("", userFilter,
-                                        new String[] { username }).getDn()
+                                        new String[] { username }).getNameInNamespace()
                                 .toString();
```

Comment:
I removed this change from the patch and kept it one issue focused. This change was causing problems in the pattern matching of user DNs so it was going to take me too far. But I would like to make a note for the future that ideally there should be consistency between the two ldap services.